### DISCUSSION TITLE
nvim 0.10.0 dev breaks inlay hints reactivation function

### DISCUSSION THREAD

**Discussion by airzocker** [mrcjkb/rustaceanvim#368] [Q&A] [answered]
In nvim 0.10.0 dev, I found a problem with the inlay hint disabling/enabling to update them after the language server is ready.
The issue lies inside the `vim.lsp.inlay_hint.enable()` function calls, which seems to have changed its signature or something.

In this file:
~/.local/share/nvim/site/pack/packer/start/rustaceanvim/lua/rustaceanvim/server_status.lua

The following code toggles inlay hints off and on to refresh:
```
vim.lsp.inlay_hint.enable(false, { bufnr = bufnr })
vim.lsp.inlay_hint.enable(true, { bufnr = bufnr })
```

But the following error occurs when opening a rust file:
```
Error executing vim.schedule lua callback: /usr/share/nvim/runtime/lua/vim/lsp/inlay_hint.lua:375: bufnr: expected number, got boolean
stack traceback:
        [C]: in function 'error'
        vim/shared.lua: in function 'validate'
        /usr/share/nvim/runtime/lua/vim/lsp/inlay_hint.lua:375: in function 'enable'
        ...er/start/rustaceanvim/lua/rustaceanvim/server_status.lua:22: in function 'handler'
        /usr/share/nvim/runtime/lua/vim/lsp/client.lua:1013: in function ''
        vim/_editor.lua: in function <vim/_editor.lua:0>
```

The relevant file (/usr/share/nvim/runtime/lua/vim/lsp/inlay_hint.lua) has this definition for the relevant `enable()` function:
```
--- @param bufnr (integer|nil) Buffer handle, or 0 or nil for current
--- @param enable (boolean|nil) true/nil to enable, false to disable
--- @since 12
function M.enable(bufnr, enable)
  vim.validate({ enable = { enable, 'boolean', true }, bufnr = { bufnr, 'number', true } })
  if enable == false then
    _disable(bufnr)
  else
    _enable(bufnr)
  end
end

```

I found a solution, which is to reverse argument order and use the `bufnr` value directly instead of a table:
```
vim.lsp.inlay_hint.enable(bufnr, false)
vim.lsp.inlay_hint.enable(bufnr, true)
```

I don't know if this direct fix would break older versions, but does anyone know where this comes from or if this should be addressed in future releases of rustaceanvim?

**Comment by mrcjkb** [comment, marked as answer]
Hey 👋 

You are using an outdated version of neovim nightly. See [the changelog](https://github.com/mrcjkb/rustaceanvim/blob/master/CHANGELOG.md#4226---2024-04-19).

**Reply by airzocker** [reply]
Hey,
thank you very much for the info, I just updated, and it works with the original code now 👍

**Reply by airzocker** [reply]
Btw, I am new to GitHub Discussions and I think your comment should be marked as the answer instead of closing the discussion, but it seems I can't do that.
